src/json.h

```diff
@@ -703,6 +706,71 @@ class JsonObject
 
         // useful debug info
         std::string line_number(); // for occasional use only
+
+        template <typename T>
```

Comment:
Probably either a new module or possibly generic_factory.
I assume you will want to re-use this functionality, so the vehicle_parts loader is out.
There is probably a decent amount of code in the different factories that can be moved to a helper module, something like this is a good first tenant.